app/src/main/java/io/homeassistant/companion/android/widgets/StaticWidget.kt

```diff
@@ -125,9 +125,16 @@ class StaticWidget : AppWidgetProvider() {
 
         if (attributeIds == null) return entity?.state
 
-        val fetchedAttributes = entity?.attributes as Map<*, *>
-        val attributeValues = attributeIds.split(",").map { id -> fetchedAttributes.get(id)?.toString() }
-        return entity.state.plus(if (attributeValues.isNotEmpty()) stateSeparator else "").plus(attributeValues.joinToString(attributeSeparator))
+        var fetchedAttributes: Map<*, *>
+        var attributeValues: List<String?>
+        try {
+            fetchedAttributes = (entity?.attributes as? Map<*, *>)!!
```

Comment:
So the `!!`  is short hand to ensure the object isn't null so that would have thrown the same/similar error.  You want to declare the map as follows:
```suggestion
            fetchedAttributes = entity?.attributes as? Map<*, *> ?: mapOf<String, String>()
```
That will set fetchedAttributes to an empty map if null.

Reply:
Oh ok that makes sense, thanks for your help :)